File: worker/src/PathManager.java
import java.util.ArrayList;

public class PathManager {
	 // Holds our products
    private static ArrayList<Product> destinationProducts = new ArrayList<Product>();

    /**
     * Adds a destination product
     * @param product
     */
    public static void addProduct(Product product) {
    	destinationProducts.add(product);
    }
    
    /**
     * Get a product
     * @param index
     * @return Product
     */
    public static Product getProduct(int index){
        return (Product)destinationProducts.get(index);
    }

    public static ArrayList<Product> getPath() { return destinationProducts; }
    
    /**
     * Get the number of destination products
     * @return Number of Products
     */
    public static int numberOfProducts(){
        return destinationProducts.size();
    }
}

File: scheduler/src/main/kotlin/genetic_algorithm/PathManager.kt
package genetic_algorithm

import java.util.*

class PathManager {
    companion object {
        // Holds our products
        private val destinationProducts = ArrayList<Product>()

        /**
         * Adds a destination product
         * @param product
         */
        fun addProduct(product: Product) {
            destinationProducts.add(product)
        }

        /**
         * Get a product
         * @param index
         * @return Product
         */
        fun getProduct(index: Int): Product? {
            return destinationProducts[index]
        }

        /**
         * Get the number of destination products
         * @return Number of Products
         */
        fun numberOfProducts(): Int {
            return destinationProducts.size
        }

        fun clear() {
            destinationProducts.clear()
        }
    }
}